lib/administration/resource/include/irods/resource_administration.hpp

```diff
@@ -176,25 +224,89 @@ namespace irods::experimental::administration
+                std::same_as<ResourceProperty, vault_path_property> ||
+                std::same_as<ResourceProperty, resource_status_property> ||
+                std::same_as<ResourceProperty, resource_comments_property> ||
+                std::same_as<ResourceProperty, resource_info_property> ||
+                std::same_as<ResourceProperty, free_space_property> ||
+                std::same_as<ResourceProperty, context_string_property>
+        auto modify_resource(RxComm& _comm, const resource_name_type _name, const ResourceProperty& _property) -> void
+        {
+            generalAdminInp_t input{};
+            input.arg0 = "modify";
+            input.arg1 = "resource";
+            input.arg2 = _name.data();
+
+            if constexpr (std::is_same_v<ResourceProperty, resource_type_property>) {
+                input.arg3 = "type";
+                input.arg4 = _property.value.data();
+            }
+            else if constexpr (std::is_same_v<ResourceProperty, host_name_property>) {
+                input.arg3 = "host";
+                input.arg4 = _property.value.data();
+            }
+            else if constexpr (std::is_same_v<ResourceProperty, vault_path_property>) {
+                input.arg3 = "path";
+                input.arg4 = _property.value.data();
+            }
+            else if constexpr (std::is_same_v<ResourceProperty, resource_status_property>) {
+                input.arg3 = "status";
+
+                if (_property.value == resource_status::up) {
+                    input.arg4 = "up";
+                }
+                else if (_property.value == resource_status::down) {
+                    input.arg4 = "down";
+                }
+                else {
+                    THROW(SYS_INVALID_INPUT_PARAM, "Invalid value for resource status");
+                }
+            }
+            else if constexpr (std::is_same_v<ResourceProperty, resource_comments_property>) {
+                input.arg3 = "comment";
+                input.arg4 = _property.value.data();
+            }
+            else if constexpr (std::is_same_v<ResourceProperty, resource_info_property>) {
+                input.arg3 = "info";
+                input.arg4 = _property.value.data();
+            }
+            else if constexpr (std::is_same_v<ResourceProperty, free_space_property>) {
+                input.arg3 = "free_space";
+                input.arg4 = _property.value.data();
+            }
+            else if constexpr (std::is_same_v<ResourceProperty, context_string_property>) {
+                input.arg3 = "context";
+                input.arg4 = _property.value.data();
+            }
+
+            if (const auto ec = rxGeneralAdmin(&_comm, &input); ec < 0) {
+                constexpr char* msg = "Could not change [{}] to [{}] for resource [{}]";
+                // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+                THROW(ec, fmt::format(msg, input.arg3, input.arg4, _name));
```

Comment:
Introduce a compile-time `to_string` function for each property type. This error message isn't very helpful without knowing what kind of property is being changed.

Reply:
Done